app/services/build_batch_from_folder_ingest.rb

```diff
@@ -4,7 +4,7 @@ class BuildBatchFromFolderIngest
               :batch_name, :batch_description
   attr_accessor :batch, :collection_repo_id
 
-  def initialize(user:, filesystem:, content_modeler:, metadata_provider:, checksum_provider:, admin_set: nil,
+  def initialize(user:, filesystem:, content_modeler:, metadata_provider: nil, checksum_provider:, admin_set: nil,
                  collection_repo_id: nil, batch_name: nil, batch_description:nil)
```

Comment:
In the interest of DRY, how about including `ActiveModel::Model` and using validation to require specific attributes?